src/Kestrel.Core/Internal/HttpConnection.cs

```diff
@@ -381,7 +377,7 @@ private HttpProtocols SelectProtocol()
             return http2Enabled && (!hasTls || Http2Id.Span.SequenceEqual(applicationProtocol.Span)) ? HttpProtocols.Http2 : HttpProtocols.Http1;
         }
 
-        public void Tick(DateTimeOffset now)
+        public void Tick(in DateTimeOffset now)
```

Comment:
`DateTimeOffset` is not a readonly struct; have added PR for it https://github.com/dotnet/coreclr/pull/19552